### ISSUE TITLE
Ambiguity between template-name and other name should be detected

### ISSUE THREAD

**Issue by e48d0f12-b0d3-485a-93ce-75b823b0ccac** [llvm/llvm-project#18860]
|  |  |
| --- | --- |
| Bugzilla Link | [18486](https://llvm.org/bz18486) |
| Version | trunk |
| OS | All |
| CC | @DougGregor,@zygoloid |

## Extended Description 
I think that sample code below should cause error because TEST is ambiguous.
```cpp
namespace FOO {
    template<int I>
    struct FOO {};
}
using namespace FOO;
void f() { FOO<0> x; }
```
However, an error does not actually occur.

If above sample is well-formed, why is next code ill-formed?
```cpp
template<int I>
struct FOO {};
namespace BAR {
    namespace FOO {
        void f() { FOO<0> x; }
    }
}
```

**Comment by Endilll** [comment]
The first example has been rejected since Clang 9: https://godbolt.org/z/EWKx3EcMP
The second example is ill-formed, and has always been rejected: https://godbolt.org/z/cb4GEjcGP
